token_dispenser_client_lib/token_dispenser_client.py

```diff
@@ -0,0 +1,197 @@
+    if not name.endswith('/'):  # Check if it's a specific parameter name
+        response = ssm.get_parameter(
+            Name=name,
+            WithDecryption=with_decryption
+        )
+    return response['Parameter']['Value']
+
+
+def get_parameters_by_path(path:str, with_decryption=True):
+    """
+    Retrieves parameters from AWS Systems Manager Parameter Store by path.
+
+    Args:
+        path (str): The path to the parameters.
+        with_decryption (bool, optional): Whether to decrypt SecureString parameters. Defaults to True.
+
+    Returns:
+        dict: A dictionary containing the parameter names and values, or None if an error occurs.
+        Returns empty dict if no parameters found on the path.
+    Raises:
+        botocore.exceptions.ClientError: If an error occurs while calling the SSM API.
+    """
+    try:
+        parameters = {}
+        next_token = None
+        while True:
+            if next_token:
+                response = ssm.get_parameters_by_path(
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption,
+                    NextToken=next_token
+                )
+            else:
+                response = ssm.get_parameters_by_path(
+                    Path=path,
+                    Recursive=True,
+                    WithDecryption=with_decryption
+                )
+
+            for param in response.get('Parameters', []):
+                parameters[param['Name']] = param['Value']
+
+            next_token = response.get('NextToken')
+            if not next_token:
+                break
+
+        return parameters
+
+    except Exception as e:
+        print(f"Error retrieving parameters from path '{path}': {e}")
+        return None
+
+
+def invoke_lambda(input_params_json, lambda_arn):
+    """
+    Invokes an AWS Lambda function with input parameters from a JSON string.
+
+    Args:
+        input_params_json (str): A JSON string containing input parameters.
+        lambda_arn (str): The ARN of the Lambda function to invoke.
+
+    Returns:
+        str: The JSON-formatted response from the Lambda function, or an error message.
+    """
+    if not lambda_arn:
+        return "Error: lambda_arn not provided"
+
+    try:
+        input_params = json.loads(input_params_json)
+    except json.JSONDecodeError:
+        return "Error: Invalid JSON"
+
+    try:
+        response = lambda_client.invoke(
+            FunctionName=lambda_arn,
+            InvocationType='RequestResponse',  # Synchronous invocation
+            Payload=json.dumps(input_params).encode('utf-8')
+        )
+
+        payload = response['Payload'].read().decode('utf-8')
+
+        # Check for function error
+        if 'FunctionError' in response:
+            return f"Error: Lambda function error: {payload}"
+
+        return payload
+
+    except Exception as e:
+        return f"Error: Lambda invocation failed: {e}"
```

Comment:
Any specific exceptions we can catch? https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda/client/invoke.html#:~:text=alias%20resolved%20to.-,Exceptions,-Lambda.Client.exceptions
(like these 2? 
Lambda.Client.exceptions.ResourceNotFoundException
Lambda.Client.exceptions.InvalidRequestContentException)